drivers/local.go

```diff
@@ -0,0 +1,139 @@
+package drivers
+
+import (
+	"strings"
+	"sync"
+
+	"github.com/livepeer/go-livepeer/net"
+)
+
+// DataDir base dir to save data to
+var DataDir string
+
+var dataCacheLen = 12
+
+type localOS struct {
+	baseURL string
+	dCache  map[string]*dataCache
+	dLock   sync.RWMutex
+}
+
+func NewLocalDriver(baseURL string) *localOS {
+	if LocalStorage != nil {
+		panic("Should only be one instance of LocalStorage driver")
+	}
+	od := &localOS{
+		baseURL: baseURL,
+		dCache:  make(map[string]*dataCache),
+		dLock:   sync.RWMutex{},
+	}
+	LocalStorage = od
+	return od
+}
+
+// EvictManifestData clears memory cache from data belonging to manifest
+func (ostore *localOS) EvictManifestData(mid string) {
+	ostore.dLock.Lock()
+	for k := range ostore.dCache {
+		if strings.HasPrefix(k, mid) {
+			delete(ostore.dCache, k)
+		}
+	}
+	ostore.dLock.Unlock()
+}
+
+func (ostore *localOS) GetData(mid, name string) []byte {
+	ostore.dLock.RLock()
+	defer ostore.dLock.RUnlock()
+	for k, cache := range ostore.dCache {
+		if strings.HasPrefix(k, mid) {
+			data := cache.GetData(name)
+			if data != nil {
+				return data
+			}
+		}
+	}
+	return nil
+}
+
+func (ostore *localOS) IsExternal() bool {
+	return false
+}
+
+func (ostore *localOS) GetInfo() *net.OSInfo {
+	return nil
+}
+
+func (ostore *localOS) SaveData(jobID int64, streamID, name string,
+	data []byte) (*net.TypedURI, error) {
+
+	var err error
+	dc := ostore.getCacheForStream(streamID)
+	dc.Insert(name, data)
+	abs := ostore.getAbsoluteURL(name)
+	turl := &net.TypedURI{
+		Storage:  net.OSInfo_LOCAL,
+		StreamID: streamID,
+		Uri:      abs,
+		Name:     name,
+	}
+	return turl, err
+}
+
+func (ostore *localOS) getCacheForStream(streamID string) *dataCache {
+	ostore.dLock.RLock()
+	sc, ok := ostore.dCache[streamID]
+	ostore.dLock.RUnlock()
+	if !ok {
+		sc = newDataCache(dataCacheLen)
+		ostore.dLock.Lock()
+		ostore.dCache[streamID] = sc
+		ostore.dLock.Unlock()
+	}
+	return sc
+}
+
+func (ostore *localOS) getAbsoluteURL(name string) string {
+	if ostore.baseURL != "" {
+		return ostore.baseURL + "/stream/" + name
+	}
+	return name
```

Comment:
Think we need to append "/stream/" unconditionally here

Reply:
why?